### ISSUE TITLE
Release tar.gz files are actually just tar files

### ISSUE THREAD

**Issue by hblanks** [pusher/oauth2_proxy#48]
The v3.0.0 release archives have `.gz` extensions, but aren't, in fact, gzipped.

## Expected Behavior

```
$ file *.gz
oauth2_proxy-v3.0.0-0-g7887272.darwin-amd64.go1.11.tar.gz:  gzip compressed data, was "oauth2_proxy-v3.0.0-0-g7887272.darwin-amd64.go1.11.tar", last modified: Thu Feb  7 21:24:45 2019, from Unix
oauth2_proxy-v3.0.0-0-g7887272.linux-amd64.go1.11.tar.gz:   gzip compressed data, was "oauth2_proxy-v3.0.0-0-g7887272.linux-amd64.go1.11.tar", last modified: Thu Feb  7 21:24:59 2019, from Unix
oauth2_proxy-v3.0.0-0-g7887272.windows-amd64.go1.11.tar.gz: gzip compressed data, was "oauth2_proxy-v3.0.0-0-g7887272.windows-amd64.go1.11.tar", last modified: Thu Feb  7 21:24:30 2019, from Unix
```

## Current Behavior

```
$ file *.gz
oauth2_proxy-v3.0.0-0-g7887272.darwin-amd64.go1.11.tar.gz:  POSIX tar archive
oauth2_proxy-v3.0.0-0-g7887272.linux-amd64.go1.11.tar.gz:   POSIX tar archive
oauth2_proxy-v3.0.0-0-g7887272.windows-amd64.go1.11.tar.gz: POSIX tar archive
```

## Possible Solution

`tar -czf` would probably do it.

## Steps to Reproduce (for bugs)

```
cd /tmp
curl -L -O https://github.com/pusher/oauth2_proxy/releases/download/v3.0.0/oauth2_proxy-v3.0.0-0-g7887272.windows-amd64.go1.11.tar.gz
curl -L -O https://github.com/pusher/oauth2_proxy/releases/download/v3.0.0/oauth2_proxy-v3.0.0-0-g7887272.darwin-amd64.go1.11.tar.gz
curl -L -O https://github.com/pusher/oauth2_proxy/releases/download/v3.0.0/oauth2_proxy-v3.0.0-0-g7887272.linux-amd64.go1.11.tar.gz
file *.gz
```

## Context

When running things like `tar -xzf ${ARCHIVE_NAME}` in an install / build script, tar fails.

## Your Environment

```
$ lsb_release -a
...
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.1 LTS
Release:        18.04
Codename:       bionic
```

**Comment by ploxiln** [comment]
just recently fixed in #43

**Comment by hblanks** [comment]
SGTM. Thank you for your work!
